### PR TITLE
parsing device out of context agent

### PR DESCRIPTION
**Pull request by eogilvy12**
### Device 

* The device can be parsed from the `context_user_agent` field.
* I think this still needs some alterations to make sure it covers as much as we want it to but this field is frequently requested for segmentation. I think there is benefit to having both a granular and categorized field but happy to discuss!

### CODE REVIEW

**Comment by jthandy** [comment]
@eogilvy12 I'm good to go on the overall business logic and agree that there are two useful levels of detail here (as you created) but I don't love the names. Can we call them `device` and `device_category`?

**Review by jthandy — CHANGES_REQUESTED** [review]
@eogilvy12 I'm good to go on the overall business logic and agree that there are two useful levels of detail here (as you created) but I don't love the names. Can we call them `device` and `device_category`?

**Comment by eogilvy12** [comment]
@jthandy ready for final review!

File: macros/cross-adapter-modeling/sessionization/segment_web_page_views__sessionized.sql [hunk 1]
```diff
@@ -133,7 +133,14 @@ session_ids as (
         
         {{dbt_utils.star(ref('segment_web_page_views'))}},
         page_view_number,
-        {{dbt_utils.surrogate_key('anonymous_id', 'session_number')}} as session_id
+        {{dbt_utils.surrogate_key('anonymous_id', 'session_number')}} as session_id,
+        case
```
Comment (jthandy) [inline macros/cross-adapter-modeling/sessionization/segment_web_page_views__sessionized.sql]: The whole point of this model (as it talks about in its docs and comments) is to create a session_id, and this logic isn't related to that. This should be in `snowplow_web_page_views`, right alongside the `device` field you just created.

**Review by jthandy — CHANGES_REQUESTED** [review]
This is very close. I just have a small code reorganization comment. Once that's done I'm good to merge.

File: macros/cross-adapter-modeling/sessionization/segment_web_page_views__sessionized.sql (hunk 1 above)
Comment (eogilvy12) [reply macros/cross-adapter-modeling/sessionization/segment_web_page_views__sessionized.sql]: @jthandy totally agree - updated!

**Review by jthandy — APPROVED** [review]

**Pull request merged**